lib/bundler/match_platform.rb

```diff
@@ -8,7 +8,7 @@ def match_platform(p)
       Gem::Platform::RUBY == platform or
       platform.nil? or p == platform or
       generic(Gem::Platform.new(platform)) == p or
-      platform === p
+      Gem::Platform.new(platform) === p
```

Comment:
Since `Gem::Platform.new(platform)` is done twice, maybe it should be wrapped in a variable?